Scripts/Prerequisite.ps1

```diff
@@ -0,0 +1,96 @@
+  if ($null -eq $rgAccess) {
+      Write-Warning "You don't have access to required resource group: $resourceGroupName. please elevate your roles via PIM and try again."
+      exit
+  }
+  Write-Host "Pre-requisites check has completed." -ForegroundColor Green
+  
+  $isKVAvailable  = $(az keyvault list --query "[?name=='$keyVaultName'] | length(@)"  -g $resourceGroupName)
+  if($isKVAvailable -eq 1)
+  {
+    Write-Host "keyvault: $keyVaultName already available ..." -ForegroundColor Green
+  }
+  else
+  {
+    Write-Host "Creating keyvault: $keyVaultName ..." -ForegroundColor Green
+    az keyvault create --name $keyVaultName --resource-group $resourceGroupName --location $location
+  }
+
+  Write-Host "Creating selfsigned certificate in keyvault ..." -ForegroundColor Green
+  az keyvault certificate get-default-policy | Out-File -Encoding utf8 defaultpolicy.json
+  az keyvault certificate create --vault-name $keyVaultName -n $certName  --policy `@defaultpolicy.json
+  az keyvault certificate download --vault-name $keyVaultName -n $certName -f cert.pem
+
+  Write-Host "Creating App registration..." -ForegroundColor Green
+  az ad app create --display-name $servicePrincipalName
+  $appId = az ad app list --display-name $servicePrincipalName --query [0].appId
+  Write-Host "Creating Service Principal..." -ForegroundColor Green
+  az ad sp create --id $appId
+
+  Write-Host "Importing certificate from keyvault to ServicePrincipal ..." -ForegroundColor Green
+  az ad sp credential reset -n $servicePrincipalName   --keyvault $keyVaultName --cert $certName --append
+   
+  Write-Host "Adding ServicePrincipal client secret to keyvault: $keyVaultName ..." -ForegroundColor Green
+  $spnResult = az ad sp credential reset --name $servicePrincipalName
+  $spnDetail = $spnResult | ConvertFrom-Json 
+  az keyvault secret set --vault-name $keyVaultName --name $keySecretForServicePrincipal --value $spnDetail.password
+  $spId = az ad sp list --display-name $servicePrincipalName --query [0].appId
+
+  Write-Host "Adding ServicePrincipal to keyvault access policies ..." -ForegroundColor Green
+  $objectId = az ad app show  --id $spId  --query 'objectId'
+  az keyvault set-policy --name $keyVaultName --object-id $objectId  --secret-permissions backup delete get list recover restore set --key-permissions backup create delete get import list recover restore sign update verify   --certificate-permissions  backup create delete deleteissuers get getissuers import list listissuers managecontacts manageissuers recover restore setissuers update
+
+  Write-Host "Creating ServiceConnection ..." -ForegroundColor Green
+  az devops service-endpoint azurerm create --azure-rm-service-principal-id  $spId  --azure-rm-subscription-id  $subscriptionId  --azure-rm-subscription-name  $subscriptionName   --azure-rm-tenant-id  $tenantId  --name $serviceConnectionName  --detect true  --azure-rm-service-principal-certificate-path  'cert.pem'  --org $organizationName  -p $projectName
```

Comment:
az devops is from azure-devops extensions, we should check this as part of pre-requisites check and if not found on user's machine and then install it before script execution.